lib/pbench/server/filetree.py

```diff
@@ -698,7 +693,7 @@ def __contains__(self, dataset: str) -> bool:
         """
         return dataset in self.datasets
 
-    def __getitem__(self, dataset: str) -> Dataset:
+    def __getitem__(self, dataset: str) -> Tarball:
```

Comment:
This change seems like it's in the wrong PR.

Reply:
No; this is just fixing a day-1 bug because I typed the wrong type. `FileTree` never returns a `Dataset` object, and what #2906 attempts is to clarify the distinction better. Interestingly, I completely missed this bug in that context.